pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -185,6 +200,91 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (rec
 		// If not,
 		//   Do nothing (as there are other virtual machines on same external IP using this volume)
 		// Call RemoveClientVmFromIPList() to remove req.NodeID to list of nodes exposed by IP referencing this file volume.
+		cnsFileVolumeClientInstance, err := cnsfilevolumeclient.GetFileVolumeClientInstance(ctx)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get CNSFileVolumeClient instance. Error: %+v", err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		err = cnsFileVolumeClientInstance.RemoveClientVMFromIPList(ctx, instance.Namespace+"/"+instance.Spec.PvcName, instance.Spec.VMName, vm.Status.VmIp)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to add VM %q with IP %q to IPList. Error: %+v", vm.Name, vm.Status.VmIp, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Successfully added VM IP %q to IPList for CnsFileAccessConfig request with name: %q on namespace: %q", vm.Status.VmIp, instance.Name, instance.Namespace)
+
+		volumeID, err := util.GetVolumeID(ctx, r.client, instance.Spec.PvcName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get volumeID from pvcName: %q. Error: %+v", instance.Spec.PvcName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		cnsVolumeID := cnstypes.CnsVolumeId{
+			Id: volumeID,
+		}
+		vSanFileShareNetPermissions := make([]vsanfstypes.VsanFileShareNetPermission, 0)
+
+		cfg, err := config.GetConfig()
```

Comment:
Can you create a dynamic client during init that can be reused across reconcile loops?

Reply:
Addressed in the PR #613